api/nakadi-event-bus-api.yaml

```diff
@@ -1503,11 +1539,39 @@ definitions:
         type: string
         default: 'random'
 
+      compatibility_mode:
+        description: |
+          Compatibility modes provides a mean for event owners to evolve their schema, given changes
+          respect the semantics defined by this field.
+
+          It's designed to be flexible enough so that producers can evolve their schemas while not
+          inadvertently breaking existent consumers.
+
+          Once defined, the compatibility mode is fixed, since otherwise it would break a predefined contract,
+          jeopardizing the value of this feature.
+
+          There are 2 compatibility modes:
+
+          - 'legacy': Schema changes are not allowed. This is used only by legacy event types that are not compatible.
+
+          - 'incompatible': Any schema modification is accepted, even if it might break existing producers or consumers.
```

Comment:
> I am not 100% sure if we should have this option.

If we create schema versioning which has to be followed by the users I do not think we have to create some kind of workarounds. If you need an incompatible schema then create new event type (btw, you event have https://github.com/zalando/nakadi/pull/464/files#diff-4492c6b26ee0a37518728ec1585420efR1553)
👎